src/python/pants/engine/legacy/parser.py

```diff
@@ -126,4 +136,32 @@ def parse(self, filepath, filecontent):
     # this juncture.
     self._parse_context._storage.clear(os.path.dirname(filepath))
     six.exec_(python, dict(self._symbols))
+
+    # Perform this check after successful execution, so we know the python is valid (and should
+    # tokenize properly!)
+    # Note that this is incredibly poor sandboxing. There are many ways to get around it.
+    # But it's sufficient to tell most users who aren't being actively malicious that they're doing
+    # something wrong, and it has a low performance overhead.
+    if self._build_file_imports_behavior != 'allow' and 'import' in python:
+      for token in tokenize.generate_tokens(StringIO(python).readline):
```

Comment:
Is this only going to read the first line of the file?

Reply:
generate_tokens takes a function to read lines; note it's `readline` not `readline()` :)